wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/Proto3WireProtocCompatibilityTests.kt

```diff
@@ -303,13 +285,10 @@ class Proto3WireProtocCompatibilityTests {
     assertThat(protocMessage.easterAnimalsPackedList).isEqualTo(listOf(EasterAnimalP3.UNRECOGNIZED, EasterAnimalP3.HEN, EasterAnimalP3.UNRECOGNIZED, EasterAnimalP3.BUNNY))
     assertThat(protocMessage.easterAnimalsPackedValueList).isEqualTo(listOf(8, 2, 9, 1))
 
-    assertThat(wireMessage.optional_easter_animal).isEqualTo(EasterAnimalK3.UNRECOGNIZED)
-    assertThat(wireMessage.identity_easter_animal).isEqualTo(EasterAnimalK3.UNRECOGNIZED)
-    assertThat(wireMessage.easter_animals_repeated).isEqualTo(listOf(EasterAnimalK3.UNRECOGNIZED, EasterAnimalK3.HEN, EasterAnimalK3.UNRECOGNIZED))
-    assertThat(wireMessage.easter_animals_packed).isEqualTo(listOf(EasterAnimalK3.UNRECOGNIZED, EasterAnimalK3.HEN, EasterAnimalK3.UNRECOGNIZED, EasterAnimalK3.BUNNY))
-    // Wire doesn't read known values so we are missing 2.
-    assertThat(easterAnimalsRepeatedReader.read(wireMessage)).isEqualTo(listOf(7, 6))
-    assertThat(easterAnimalsPackedReader.read(wireMessage)).isEqualTo(listOf(8, 9))
+    assertThat(wireMessage.optional_easter_animal).isEqualTo(EasterAnimalK3.Unrecognized(5))
+    assertThat(wireMessage.identity_easter_animal).isEqualTo(EasterAnimalK3.Unrecognized(6))
+    assertThat(wireMessage.easter_animals_repeated).isEqualTo(listOf(EasterAnimalK3.Unrecognized(7), EasterAnimalK3.HEN, EasterAnimalK3.Unrecognized(6)))
```

Comment:
Very nice here cause we keep the order and all the data lives in the same place

Reply:
YES!